TradeItIosTicketSDK2/TradeItMarketService.swift

```diff
@@ -6,93 +6,47 @@
-        )
-
-        // TODO: Fix this. Our connector doesn't support a way to just get back a string array from JSON.
-        self.marketDataService.connector.sendEMSRequest(request, withCompletionBlock: { result, jsonResponse in
-            if let data = jsonResponse?.data(using: String.Encoding.utf8.rawValue),
-                let symbolsTemp = try? JSONSerialization.jsonObject(with: data, options: []) as? [String],
-                let symbols = symbolsTemp {
-                onSuccess(symbols)
-            }  else {
-                let jsonString = jsonResponse as String?
-                let errorResult = TradeItRequestResultFactory.build(TradeItErrorResult(), jsonString: jsonString) as? TradeItErrorResult
-                onFailure(errorResult ?? TradeItErrorResult.error(withSystemMessage: "Failed to fetch FX symbols"))
-            }
-        })
-    }
-}
-
 @objc public class TradeItMarketService: NSObject, MarketDataService {
-    let marketDataService: TradeItMarketDataService
+    let connector: TradeItConnector
 
-    init(apiKey: String, environment: TradeitEmsEnvironments) {
-        let connector = TradeItConnector(apiKey: apiKey, environment: environment, version: TradeItEmsApiVersion_2)
-        self.marketDataService = TradeItMarketDataService(connector: connector)
+    init(connector: TradeItConnector) {
+        self.connector = connector
     }
 
     public func getQuote(
         symbol: String,
         onSuccess: @escaping (TradeItQuote) -> Void,
         onFailure: @escaping (TradeItErrorResult) -> Void
     ) {
-        let quotesRequest = TradeItQuotesRequest(symbol: symbol)
+        let quoteRequest = TradeItQuotesRequest(symbol: symbol)
 
-        self.getQuote(quoteRequest: quotesRequest, onSuccess: onSuccess, onFailure: onFailure)
-    }
+        let request = TradeItRequestResultFactory.buildJsonRequest(
+            for: quoteRequest,
+            emsAction: getEndpoint(forRequest: quoteRequest),
+            environment: self.connector.environment
+        )
 
-    private func getQuote(
-        quoteRequest: TradeItQuotesRequest,
-        onSuccess: @escaping (TradeItQuote) -> Void,
-        onFailure: @escaping (TradeItErrorResult) -> Void
-    ) {
-        self.marketDataService.getQuoteData(
-            quoteRequest,
-            withCompletionBlock: { result in
-                if let quotesResult = result as? TradeItQuotesResult,
-                    let quote = quotesResult.quotes?.first as? TradeItQuote {
-                    onSuccess(quote)
-                } else if let errorResult = result as? TradeItErrorResult {
-                    onFailure(errorResult)
-                } else {
-                    onFailure(TradeItErrorResult(title: "Market Data failed", message: "Fetching the quote failed. Please try again later."))
-                }
+        self.connector.send(request, targetClassType: TradeItQuotesResult.self, withCompletionBlock: { result in
```

Comment:
TODO: Make sure this still works. I saw the market data missing on the trading screen.